android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -254,11 +256,15 @@ constructor(
   suspend fun <R : Resource> addOrUpdate(addMandatoryTags: Boolean = true, resource: R) {
     resource.updateLastUpdated()
     try {
-      fhirEngine.get(resource.resourceType, resource.logicalId).run {
-        val updateFrom = updateFrom(resource)
-        fhirEngine.update(updateFrom)
+      val previousResourceCopy = fhirEngine.get(resource.resourceType, resource.logicalId)
+      fhirEngine.update(resource)
```

Comment:
Is this still required? `fhirEngine.update(processedUpdatedFromResource)` is called online 267.